Pod/Classes/NWSTokenView.swift

```diff
@@ -365,28 +366,28 @@ open class NWSTokenView: UIView, UIScrollViewDelegate, UITextViewDelegate
             self.label.frame = CGRect(x: x, y: y, width: self.label.bounds.width-self.tokenViewInsets.left-self.tokenViewInsets.right, height: self.labelMinimumHeight)
             self.label.sizeToFit()
             // Reset frame after sizeToFit
-            self.label.frame = CGRect(x: x, y: y, width: self.label.bounds.width, height: self.labelMinimumHeight)
+            let height = max(self.label.bounds.height, self.textViewMinimumHeight)
+            self.label.frame = CGRect(x: x, y: y, width: self.label.bounds.width, height: height)
             x += self.label.bounds.width
             remainingWidth -= x
         }
-        
     }
     
     private var emptyTextViewFrame = CGRect.zero
 
-    // TODO: Figure out a non-magic way of fixing vertical alignment.
-    private let magicYOffset: CGFloat = 3
-
     /// Sets up token view text view.
     fileprivate func setupTextView(offsetX x: inout CGFloat, offsetY y: inout CGFloat, remainingWidth: inout CGFloat)
     {
         // Set placeholder text (ignore if tokens exist, text exists, or is currently active field)
         self.placeholder.isHidden = self.tokens.count > 0 || self.textView.text.count > 0
 
+        self.textView.sizeToFit()
+        let height = max(self.textView.bounds.height, max(self.label.bounds.height, max(self.textViewMinimumHeight, self.tokenHeight)))
```

Comment:
I wasn't sure Swift had this, but it does!

```suggestion
        let height = max(self.textView.bounds.height, self.label.bounds.height, self.textViewMinimumHeight, self.tokenHeight)

```

Reply:
Wonder when they added because I'm sure I tried before and it didn't work. Thanks for checking, way better this way.